images/encryption/encryption.go

```diff
@@ -0,0 +1,299 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package encryption
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"encoding/json"
+	"io"
+	"strings"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/pkg/errors"
+)
+
+// LayerInfo holds information about an image layer
+type LayerInfo struct {
+	// The Id of the layer starting at 0
+	ID         uint32
```

Comment:
I keep seeing this: why are integer ids being used to identify layers?

Reply:
I can call it the `Number` of the layer rather than `ID`? It is used in the first column in the following:
```
#> ./bin/ctr images layerinfo  --layer 1 --layer 4 docker.io/library/ubuntu:latest
   #                                                                    DIGEST      PLATFORM   SIZE   ENCRYPTION   RECIPIENTS
   1   sha256:9d866f8bde2a0d607a6d17edc0fbd5e00b58306efc2b0a57e0ba72f269e7c6be   linux/amd64    849                          
   4   sha256:afde35469481d2bc446d649a7a3d099147bbf7696b66333e76a411686b617ea1   linux/amd64    163                          

```

Reply:
Renamed to `Index`.